File: Sliding-Window/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums, pivot):
        less = 0
        equal = 0

        # Count elements less than and equal to pivot
        for num in nums:
            if num == pivot:
                equal += 1
            elif num < pivot:
                less += 1

        start = 0
        last = less + equal
        ans = [pivot] * len(nums)  # Initialize ans array with pivot

        # Distribute elements in ans array
        for num in nums:
            if num < pivot:
                ans[start] = num
                start += 1
            elif num > pivot:
                ans[last] = num
                last += 1

        return ans

File: Sliding-Window/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int less = 0, equal = 0;

        for(int i:nums){
            if(i == pivot) equal++;
            else if(i < pivot) less++;
        }

        int start = 0;
        int last = less + equal;
        vector<int> ans(nums.size(), pivot);
        
        for(int i:nums){
            if(i < pivot){
                ans[start] = i;
                start++;
            }
            else if(i > pivot){
                ans[last] = i;
                last++;
            }
        }

        return ans;
    }
};